crates/ruff_linter/src/rules/flake8_type_checking/snapshots/ruff_linter__rules__flake8_type_checking__tests__quote_typing-only-third-party-import_quote.py.snap

```diff
@@ -1,399 +1,504 @@
+quote.py:99:35: TCH002 [*] Move third-party import `pandas.Series` into a type-checking block
+    |
+ 98 | def f():
+ 99 |     from pandas import DataFrame, Series
+    |                                   ^^^^^^ TCH002
+100 | 
+101 |     def func(self) -> DataFrame | list[Series]:
+    |
+    = help: Move into type-checking block
 
 ℹ Unsafe fix
-    1   |+from typing import TYPE_CHECKING
+1       |-from typing import Literal
+    1   |+from typing import Literal, TYPE_CHECKING
     2   |+
     3   |+if TYPE_CHECKING:
     4   |+    from pandas import DataFrame, Series
-1   5   | def f():
-2   6   |     from pandas import DataFrame
-3   7   | 
+2   5   | 
+3   6   | 
+4   7   | def f():
 --------------------------------------------------------------------------------
-93  97  | 
-94  98  | 
-95  99  | def f():
-96      |-    from pandas import DataFrame, Series
+96  99  | 
 97  100 | 
-98      |-    def func(self) -> DataFrame | list[Series]:
-    101 |+    def func(self) -> "DataFrame | list[Series]":
-99  102 |         pass
+98  101 | def f():
+99      |-    from pandas import DataFrame, Series
+100 102 | 
+101     |-    def func(self) -> DataFrame | list[Series]:
+    103 |+    def func(self) -> "DataFrame | list[Series]":
+102 104 |         pass
+103 105 | 
+104 106 | def f():
+
+quote.py:105:44: TCH002 [*] Move third-party import `django.contrib.auth.models.AbstractBaseUser` into a type-checking block
+    |
+104 | def f():
+105 |     from django.contrib.auth.models import AbstractBaseUser
+    |                                            ^^^^^^^^^^^^^^^^ TCH002
+106 |     from typing import Annotated
+    |
+    = help: Move into type-checking block
+
+ℹ Unsafe fix
+1       |-from typing import Literal
+    1   |+from typing import Literal, TYPE_CHECKING
+    2   |+
+    3   |+if TYPE_CHECKING:
+    4   |+    from django.contrib.auth.models import AbstractBaseUser
+2   5   | 
+3   6   | 
+4   7   | def f():
+--------------------------------------------------------------------------------
+102 105 |         pass
+103 106 | 
+104 107 | def f():
+105     |-    from django.contrib.auth.models import AbstractBaseUser
+106 108 |     from typing import Annotated
+107 109 | 
+108     |-    def foo(self, user: AbstractBaseUser["int"], view: "type[CondorBaseViewSet]"):
+    110 |+    def foo(self, user: "AbstractBaseUser[int]", view: "type[CondorBaseViewSet]"):
+109 111 |         pass
+110 112 | 
+111     |-    def foo(self, user: AbstractBaseUser['int']):
+    113 |+    def foo(self, user: "AbstractBaseUser[int]"):
+112 114 |         pass
+113 115 | 
+114     |-    def foo(self, user: AbstractBaseUser['int', "str"]):
+    116 |+    def foo(self, user: "AbstractBaseUser[int, str]"):
+115 117 |         pass
+116 118 | 
+117     |-    def set_role(self, type1: AbstractBaseUser[Literal["user", "admin"], Annotated["int", "1", 2]]):
+    119 |+    def set_role(self, type1: "AbstractBaseUser[Literal['user', 'admin'], Annotated['int', 1, 2]]"):
```

Comment:
This doesn't seem correct, changing `"1"` to `1` in the `Annotated` type

Reply:
I think this is because in the catch all case within the visitor, you're not checking for `AnnotatedNonFirstElm`